tests/zkevm/test_worst_compute.py

```diff
@@ -82,22 +83,19 @@ def make_dup(index: int) -> Opcode:
 def test_worst_zero_param(
     state_test: StateTestFiller,
     pre: Alloc,
-    fork: Fork,
     opcode: Op,
 ):
     """Test running a block with as many zero-parameter opcodes as possible."""
     env = Environment()
-    max_code_size = fork.max_code_size()
 
-    code_prefix = Op.JUMPDEST
-    iter_loop = Op.POP(opcode)
-    code_suffix = Op.PUSH0 + Op.JUMP
-    code_iter_len = (max_code_size - len(code_prefix) - len(code_suffix)) // len(iter_loop)
-    code = code_prefix + iter_loop * code_iter_len + code_suffix
-    assert len(code) <= max_code_size
+    opcode_sequence = opcode * MAX_STACK_HEIGHT
+    target_contract_address = pre.deploy_contract(code=opcode_sequence)
+
+    caller_contract_code = While(body=Op.POP(Op.STATICCALL(address=target_contract_address)))
```

Comment:
```suggestion
    caller_contract_code = (
        Op.JUMPDEST + Op.JUMPDEST + Op.STATICCALL(address=target_contract_address) + Op.JUMP
    )
```
Tiny optimization here because we can actually use the return code of the staticcall as the jump destination.

Reply:
There's another optimization possible, which is what we do in many other tests -- put as many iterations of STATICCALL as possible considering the max contract size and jump to the beginning at the end.

As in:
```
JUMPDEST
POP(STATICCALL)
POP(STATICCALL)
...
...
JUMP 0
```

So we amortize the 8 gas of JUMP in as many STATICCALL as possible. It is basically recovering the deleted lines in this PR but framing it for the STATICCALL instead of `opcode`.

Reply:
I like the idea of using `DUP1` instead of a `PUSH` operation.

Using `PUSH20` to push the address consumes 20 bytes of immediate value and costs 3 gas, whereas `DUP1` only costs 2 gas and requires no immediate data.

@marioevz  I think this could further be optimized in the current `While` implementation, so I might not change my implementation here, wdyt.

Reply:
Nvm I need to get some coffee and then I should revisit above post, because it is not correct.

Reply:
Thank you for the feedback! I’ve summarized our discussion in [this comment](https://github.com/ethereum/execution-spec-tests/issues/1704#issuecomment-2975867921). I was thinking we could merge this PR first, and then follow up with a separate one to finalize the implementation and update all related usages?

Reply:
I found an existing helper function that is more optimized, let me update the code with the helper